core/src/main/scala/weaponregex/parser/Parser.scala

```diff
@@ -49,6 +49,10 @@ abstract class Parser(val pattern: String) {
     */
   val predefCharClassChars: String
 
+  /** Minimum number of character class items of a valid character class
+    */
+  val minCharClassItem: Int
```

Comment:
These are all the platform-specific things that a flavor does, right? Perhaps it makes sense to extract these into 1 single value like a `PlatformOptions` that a parser has to define (or is passed)?

Reply:
They are, but those values alone are not complete. I tried to parameterize the differences as much as I can, but those are just the simplest cases. Some differences require modifications to intermediate rules. More severe cases need separate definitions of a parsing rule. So, compressing all the differences into a single value/object is not really feasible.